openwisp_monitoring/device/tests/test_api.py

```diff
@@ -242,6 +243,16 @@ def test_garbage_clients(self):
         )
         self.assertEqual(r.status_code, 400)
 
+    def test_no_resources(self):
+        o = self._create_org()
+        d = self._create_device(organization=o)
+        data = deepcopy(self._data())
```

Comment:
deepcopy is not needed here because:

```python
def _data(self):
        return {
```

`return` always creates a new data structure.